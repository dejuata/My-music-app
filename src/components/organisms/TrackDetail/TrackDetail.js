import React from 'react'

export const TrackDetail = () => {
    return (
        <div>
            <p>Track Detail</p>
        </div>
    )
}
