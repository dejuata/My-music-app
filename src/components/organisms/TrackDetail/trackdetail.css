.track-detail {
    width: 100%;
    padding: 12px;
    box-sizing: border-box;
}

.track-detail__image {
    margin: 0;
    /* padding: 20px; */
    /* box-sizing: border-box; */
}

.track-detail__image img{
    width: 100%;
    object-fit: cover;
    border-radius: 12px;
}

.track-detail__body {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    /* justify-content: space-between; */
    margin-top: 24px;
    row-gap: 20px;
}

.track-detail__body-title h2 {
    margin: 0 ;
    font-size: var(--xlg);
    font-weight: bold;
}

.track-detail__body-title p {
    margin: 0;
    font-size: var(--lg);
}

/* .track-detail__body-title p:nth-child(2) {
    margin-bottom: 32px;
    

} */

.track-detail__body-info p {
    margin: 0;
}

.track-detail__body-info p:nth-child(1) {
    font-size: var(--md);
    text-align: justify;
    font-weight: 300;
    margin-bottom: 12px;

}

.track-detail__body-info p:nth-child(2) {
    /* margin-bottom: 32px; */
    font-weight: 300;
    font-size: var(--sm);
}

.track-detail__body-footer {
    width: 100%;
}

@media (min-width: 768px) {

    .track-detail {
        display: flex;
        justify-content: center;
        column-gap: 24px;
    }

    .track-detail__image {
        padding: 0;
    }

    .track-detail__body {
        margin-top: 0;
        justify-content: space-between;
    }
}