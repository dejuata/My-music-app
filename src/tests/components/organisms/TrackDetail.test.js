import { MemoryRouter } from "react-router-dom";
import { TrackDetail } from "../../../components/organisms/TrackDetail/TrackDetail";
import { demoTracks } from "../../fixtures/demoTracks";
import { mount } from 'enzyme';

describe('Test TrackDetail', () => {
    const track = demoTracks.tracks[0];

    const wrapper = mount(
        <MemoryRouter>
            <TrackDetail
                { ...track }
                />
        </MemoryRouter>
    )

    test('Should be render TrackDetail', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Should be render <SpotifyButton />', () => {

        expect( wrapper.find('SpotifyButton').exists() ).toBe(true);

    })
})