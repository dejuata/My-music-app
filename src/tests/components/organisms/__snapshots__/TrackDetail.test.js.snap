// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TrackDetail Should be render TrackDetail 1`] = `
<div
  className="o-track-detail"
>
  <figure
    className="o-track-detail__image"
  >
    <img
      alt="Voy Pa'Encima"
      src="https://i.scdn.co/image/ab67616d00001e021386b29ab009dd27c398d141"
    />
  </figure>
  <div
    className="o-track-detail__body"
  >
    <div
      className="o-track-detail__body-title"
    >
      <h2>
        Desn√∫date Mujer
      </h2>
      <p>
        Frankie Ruiz
      </p>
    </div>
    <div
      className="o-track-detail__body-info"
    >
      <p>
        This song belongs to the Album 
        <strong>
          Voy Pa'Encima
        </strong>
         released on 
        .
      </p>
      <p>
        <strong>
          Duration: 
        </strong>
        4:47
         min.
      </p>
    </div>
    <div
      className="o-track-detail__body-footer"
    >
      <a
        href="https://url-track"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          className="btn btn-secondary"
        >
          Listen on Spotify
        </div>
      </a>
    </div>
  </div>
</div>
`;
